lib/spack/spack/database.py

```diff
@@ -1071,6 +1071,8 @@ def _add(
                 "Specs added to DB must be concrete.")
 
         key = spec.dag_hash()
+        spec_run_hash = spec._runtime_hash
+        spec_pkg_hash = spec._package_hash
```

Comment:
Hmm, it might be fine to use the methods from here instead of accessing the properties directly.  Or, to really make sure we're not re-computing, how would you feel about, e.g. `getattr(spec, ht.runtime_hash.attr, None)`?

Reply:
I'm actually having a hard time thinking of why we need the runtime hash anymore.  We could always add it back if we need it, but why do we want to keep it on the spec here?  I'm tempted to say we should just remove it.